### ISSUE TITLE
Deleted topics

### ISSUE THREAD

**Issue by tkellogg** [mjkoster/I-D#4]
What happens when a client deletes a topic that another client is subscribed to?

* Do queued messages get lost?
* Is the DELETE successful?
* Can a subscriber veto the DELETE?

**Comment by mjkoster** [comment]
There is the bigger question of queueing. I'd suggest we don't define queueing, only require storage of the last published value.

The assumption is that the REMOVE operation requires system-wide agreement beforehand. We aren't trying to prevent malicious clients from deleting topics through any mechanism other than access control methods that are already known. Likewise, I don't think we are building in a method for all clients to agree through the CoAP PubSub protocol. Usually these are handled using a higher level protocol. One simple use case is a single sensor publishing on it's own topic. It could REMOVE the topic

Since a topic is ultimately a CoAP resource, I think we are pretty much bound to follow CoAP in terms of what happens to in-flight message ACKs on deleted resources, etc. It's a question that might be worth clarifying in the draft.

Do you have use case requirements that would inform some specific behavior?

**Comment by mjkoster** [comment]
To clarify the queueing question, we don't want to exclude the possibility of queueing, and certainly for guaranteed delivery of messages like the at-least-once service class, queueing will be needed. The point is more that we only want to specify the interface protocol, not what's inside the broker, as much as possible to provide interoperability. The last value storage needs to be specified for REST API compatibility. The queueing may not really be optional in a practical sense, but it's good to keep definitions of internals out of the interface specification. Maybe a best practices section would be a good place to write more about use of internal message storage.

**Comment by tkellogg** [comment]
Agreed. It would be much cleaner if words like "queueing" we're avoided.
Let's see if we can pull it off...
